File: src/main/java/com/jaffa/rpc/lib/entities/RequestContextHelper.java
package com.jaffa.rpc.lib.entities;

import com.jaffa.rpc.lib.security.SecurityTicket;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class RequestContextHelper {

    private static final ThreadLocal<String> sourceModuleId = new ThreadLocal<>();
    private static final ThreadLocal<SecurityTicket> securityTicketThreadLocal = new ThreadLocal<>();

    public static String getSourceModuleId() {
        return sourceModuleId.get();
    }

    public static void setSourceModuleId(String sourceModuleId) {
        RequestContextHelper.sourceModuleId.set(sourceModuleId);
    }

    public static SecurityTicket getTicket() {
        return securityTicketThreadLocal.get();
    }

    public static void setSecurityTicket(SecurityTicket securityTicket) {
        RequestContextHelper.securityTicketThreadLocal.set(securityTicket);
    }

    public static void setMetaData(@NotNull Command command) {
        setSourceModuleId(command.getSourceModuleId());
        setSecurityTicket(command.getTicket());
    }

    public static void removeMetaData() {
        sourceModuleId.remove();
        securityTicketThreadLocal.remove();
    }
}

File: src/main/kotlin/com/jaffa/rpc/lib/common/RequestContextHelper.kt
package com.jaffa.rpc.lib.common

import com.jaffa.rpc.lib.entities.Command
import com.jaffa.rpc.lib.security.SecurityTicket

object RequestContextHelper {

    private val sourceModuleId = ThreadLocal<String>()

    private val securityTicketThreadLocal = ThreadLocal<SecurityTicket>()

    @kotlin.jvm.JvmStatic
    fun getSourceModuleId(): String {
        return sourceModuleId.get()
    }

    @kotlin.jvm.JvmStatic
    val ticket: SecurityTicket?
        get() = securityTicketThreadLocal.get()

    fun setMetaData(command: Command) {
        sourceModuleId.set(command.sourceModuleId)
        securityTicketThreadLocal.set(command.ticket)
    }

    fun removeMetaData() {
        sourceModuleId.remove()
        securityTicketThreadLocal.remove()
    }
}